src/portalnet/socket.rs

```diff
@@ -44,3 +49,19 @@ fn find_assigned_ip() -> Option<IpAddr> {
     }
     None
 }
+
+#[cfg(windows)]
+fn find_assigned_ip() -> Option<IpAddr> {
+    let adapters = ipconfig::get_adapters().unwrap();
```

Comment:
Presumably unwrap is panicking here on a failure. I think we want to return `None` instead.